### DISCUSSION TITLE
[Idea] When requesting a password reset on an inexistent account, send an email to inform the user

### DISCUSSION THREAD

**Discussion by lmeyer1** [PrestaShop/PrestaShop#34785] [Ideas]
### The problem

When a customer asks to reset the password, but the account does not exist, the user is left without information.
Maybe the customer registered with another email, and he doesn't remember. He will not find the expected mail in his inbox, neither in the spam folder, and he will contact customer service. Or he will abandon and order in a concurrent shop.

### The solution

If we would send an email to the customer to tell him that no account with this email exists, he would understand immediately that his account used one of his alternative email addresses.
He will not bother customer service, and be able to pass an order immediately.

### Do you plan to work on this subject?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by matks** [comment]
I'm afraid that would be allowing user enumeration https://www.hacksplaining.com/prevention/user-enumeration ?

If the "reset password" behaves differently if email inserted exists or not, using the "reset password" page would allow a hacker to know if user is registered or not

**Comment by lmeyer1** [comment]
@matks I am fully aware of this problem. By designing the new solution, particular care must be given to this problem.

The solution I recommend does not allow user enumeration in any way and is inspired for Password reset email best practices as layed out here https://postmarkapp.com/guides/password-reset-email-best-practices.

Specifically, the visible part of the website must behave exactly the same way if the account exists or if it does not exist. Only the owner of the mailbox will receive the email that tells him that the account does not exist.
Thus the hacker would need access to all the mailboxes for which he would like to do user enumeration, a condition that realistically cannot ever be met. If the hacker would have access to the user's mailbox, he could always tell if the user had an account, because he would receive or not receive the password reset mail.

Therefore the solution I recommend ***does not reduce in any way the security of the website***, but it does increase the clarity of the communication with the user:
- the problem with mails going to the spam folder is mitigated, because if the user does not receive a mail, this is the only possible reason. The possibility of using a different email is eliminated.
- the user has no doubt that he doesn't have an account yet
    - his account could be associated with a different e-mail address
    - he is confusing this website with another one where he has an account (in our use case, the website of France and the website of Switzerland do not share their user base).

**Reply by matks** [reply]
Aaaaah I understand now.

> If we would send an email to the customer to tell him that no account with this email exists, he would understand immediately that his account used one of his alternative email addresses.

But I see another flaw in this solution (sorry 😄 I'm not doing this just to annoy you, just challenging the idea) : it allows to spam someone. For example if I don't like john-doe@gmail.com I can use the "reset password" feature with john-doe@gmail.com input a thousand times and he will receive a thousand emails

But maybe I am overthinking 😅 wdyt @MatShir

**Comment by lmeyer1** [comment]
When an account is inactive or when the user recently changed his password, then the password reset page shows an error instead of sending silently the error to the user.

The errors are 
- You cannot regenerate the password for this account.
- You can regenerate your password only every %d minute(s)

@matks This ***IS*** information disclosure. It does not allow user enumeration, because it applies only to a few accounts, but it is still a problem.

**Reply by matks** [reply]
Then I guess we need to fix that 😄 

cc @matthieu-rolland
